lai_s3/s3.py

```diff
@@ -117,18 +125,82 @@ def _upload_file(
             self,
             bucket: str,
             object: str,
-            filename: Union[L.storage.Path, str]
+            filename: Union[lapp.storage.Path, str]
     ):
         with open(filename, 'rb') as _f:
             self.resource.meta.client.upload_fileobj(
                 Fileobj=_f, Bucket=bucket, Key=object
             )
 
     def run(self, action, *args, **kwargs):
-
         if action == "get_filelist":
             self._get_filelist(*args, **kwargs)
         elif action == "download_file":
             self._download_file(*args, **kwargs)
         elif action == "upload_file":
             self._upload_file(*args, **kwargs)
+        elif action == 'create_dataset':
+            return self._create_dataset(*args, **kwargs)
+
+
+    def _get_s3_items(data, idx, transforms=None, data_type='img', label_map=None):
```

Comment:
This looks more like something that is better managed outside of this class.  There's a few assumptions that aren't obvious as a user. For example, image data stored on s3 needs to follow the storage hierarchy <>/label/file.jpg.  It's not clear that this needs to be the case.

Reply:
It is a default setting that I believe is overwriteable and is explained in the doc strings in the S3 class. I can move the explanation up to here. Answering from my phone.